### PR TITLE
bug fix in +nap:to

### LINKED ISSUE #1100: Possible bug removing the head of a qeu
It seems that the logic for removing the head of a queue doesn't behave as it should:

```
> =a (~(gas to `(qeu @)`~) [4 3 2 1 ~])
> [`(qeu @)`a ~(top to a)]
[{1 2 3 4} [~ u=4]]
> =a ~(nap to a)
> [`(qeu @)`a ~(top to a)]
[{1 2 3} [~ u=3]]
> =a ~(nap to a)
> [`(qeu @)`a ~(top to a)]
[{1 3} [~ u=3]]
> =a ~(nap to a)
> [`(qeu @)`a ~(top to a)]
[{3} [~ u=3]]
> =a ~(nap to a)
> [`(qeu @)`a ~(top to a)]
[{} ~]
```
I thought this might have been an issue with the pretty-printer, but this might the balancing arm acting up.

If we have this tree
```
          2
        /   \
      1       3
```
The head is 3, so far so good.
```
> =a (~(gas to `(qeu @)`~) [3 2 1 ~])
> a
[n=2 l={1} r={3}]
> ~(top to a)
[~ u=3]
```
but if we remove the head:
```
> =a ~(nap to a)
> a
[n=1 l={} r={3}]
> ~(top to a)
[~ 3]
```
we get this (?):
```
          1
        /   \
      ~       3
```
This could be where the problem begins:

[bal(a ^+(a [p.b q.b r.a]))](https://github.com/urbit/arvo/blob/master/sys/hoon.hoon#L1871)

If instead of using the right branch of the tree ([p.b q.b **r.a**], replicating the head) 
we use the node ([p.b q.b **[n.a ~ ~]**)
```
> a
[n=2 l={1} r={3}]
> =b ~(get to ?~(a ~ l.a))
> b
[p=1 q={}]
> ~(bal to ^+(a [p.b q.b [?~(a ~ n.a) ~ ~]]))
[n=2 l=[n=1 l={} r={}] r=~]
> `(qeu @)`~(bal to ^+(a [p.b q.b [?~(a ~ n.a) ~ ~]]))
{1 2}
> ~(top to `(qeu @)`~(bal to ^+(a [p.b q.b [?~(a ~ n.a) ~ ~]])))
[~ u=2]
```
we get this tree:
```
          2
        /   \
      1      ~
```

I'm having some issues introducing this changes in hoon.hoon:
```
[%hoon-compile ~raplep-nammeb]
ride-parsing
ride-compiling
-find.cain
```
but at least it seems that something is wrong with the qeu, if I'm not mistaken.

### PR DESCRIPTION
**Pull request by yosoyubik**
Fixes #1100 (a bug in +nap:to affecting three-item queues)
Unit tests added in /tests/sys/hoon/qeu

### CODE REVIEW

File: tests/sys/hoon/qeu.hoon [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    !>  (need ~(top to queue))
+    !>  i.test
+--
+::
+|%
+::
+++  test-descending-sorted-list  ^-  tang
+  ::
+  =+  queue=(~(gas to `(qeu @)`~) descending)
+  ;:(welp (unqueue [queue descending]))
```
[the remark this answers is not in the document]
Comment (yosoyubik) [reply tests/sys/hoon/qeu.hoon]: Done!

**Comment by ixv** [comment]
>The queue bug fixed by this PR could be the cause of our long-standing one-way connectivity issues
Woah

It looks like this function is only used twice: in ames and eyre. Seems unlikely that anything is depending on its brokenness, hotfix is probably a good idea.

**Comment by joemfb** [comment]
Agreed, the right thing is to edit a PR to target the appropriate branch. It seems like `hotfix` is appropriate in this case. Thanks, @yosoyubik!

**Comment by yosoyubik** [comment]
Thank you guys! I didn't expect this to be useful at all... There I was just making a small tic-tac-toe app and look at this :D 

I have just edited the PR to target hotfix, let me know if something else is needed.

**Pull request merged**